tools/SourceKit/lib/SwiftLang/SwiftLangSupport.cpp

```diff
@@ -933,11 +933,18 @@ std::string SwiftLangSupport::resolvePathSymlinks(StringRef FilePath) {
 
   DWORD numChars = GetFinalPathNameByHandleW(fileHandle, full_path, MAX_PATH,
                                             FILE_NAME_NORMALIZED);
+  // GetFinalPathNameByHandleW returns a NTPath (e.g. \\?\C:\Users\Swift) which
+  // we transform into a traditional DOS path by dropping the \\?\.
+  int offset = numChars > 4 && full_path[0] == L'\\' && full_path[1] == L'\\' &&
+       full_path[2] == L'?' && full_path[3] == L'\\'
+          ? 4
+          : 0;
+  numChars -= offset;
```

Comment:
This seems … worrisome.  What if the path returned is some other kernel path?  e.g. `\\?\Device\Harddisk0\…`  Why not do something like `NtOpenFile`, `GetFinalPathNameByHandle`?

Reply:
I've changed this to query for VOLUME_NAME_DOS which still gives a `\\?\` prefixed path, but that should mean we never encounter an NT Path like `\Device\HarddiskX`